### PR TITLE
feat(cli): allow to specify local directory as a chart-path

### LINKED ISSUE #4325: keptn install: Allow to specify local directory as a chart-path
Right now it's a huge pain to install Keptn from a draft release / temporary build, e.g.:

```console
download keptn-installer-0.8.4-dev.tgz # download or build helm charts
python3 -m http.server &
keptn install --chart-repo=http://0.0.0.0:8000/keptn-installer-0.8.4-dev.tgz
killall python3
```

This could be much easier, e.g., by being able to specify a chart repo on the local filesystem, e.g.:

```console
download keptn-installer-0.8.4-dev.tgz # download or build helm charts
keptn install --chart-repo=./dist/keptn-installer-0.8.4-dev.tgz
```

This change would have to be done in the helper functions used here:
https://github.com/keptn/keptn/blob/49022062c4ab1c63d8b93794aa810953629936a2/cli/cmd/install.go#L203-L213
https://github.com/keptn/keptn/blob/49022062c4ab1c63d8b93794aa810953629936a2/cli/pkg/helm/helm_helper.go#L66-L84

# Definition of Done

- [ ] Keptn can be installed/upgraded using a chart repo from local disk space by using the `--chart-repo` flag

# Acceptance Criteria

I can build a local version of helm charts from master branch, using
```
./gh-actions-scripts/build_helm_charts.sh 0.8.4-dev 0.8.4-dev 0.8.3
```

and install it afterwards using
```
keptn install --chart-repo=./keptn-charts/keptn-0.8.4-dev.tgz
```

### PR DESCRIPTION
**Pull request by jimil749**
Signed-off-by: Jimil Desai <jimildesai42@gmail.com>

## This PR

- Allows users to install Keptn from a draft release by pointing to a chart repo on the local filesystem, eg:
```
download keptn-installer-0.8.4-dev.tgz # download or build helm charts
keptn install --chart-repo=./dist/keptn-installer-0.8.4-dev.tgz
```

### Related Issues

Fixes #4325 

### How to test

1. Download Keptn
2. Install Keptn by pointing to chart repo in the local filesystem: `keptn install --chart-repo=./keptn-0.8.7.tgz`

### CODE REVIEW

File: cli/cmd/install.go [hunk 1]
```diff
@@ -196,14 +197,35 @@ type InstallCmdHandler struct {
 	userInput        common.IUserInput
 }
 
+func isValidURL(chartURL string) bool {
```
Comment (warber) [inline cli/cmd/install.go:200]: pls use the func from keptn go-utils https://github.com/keptn/go-utils/blob/ef732acee71f359e7c5a591a0927e489edd242a5/pkg/common/httputils/httputils.go#L63

**Review by warber — CHANGES_REQUESTED** [review]

File: cli/cmd/install.go (hunk 1 above)
Comment (jimil749) [reply cli/cmd/install.go:200]: Ahh ok, I was not aware of that! Will do that. :+1:

**Comment by warber** [comment]
Hi, @jimil749  thanks so far, your solution works :+1: 
However, the installation fails when trying to install keptn using the `continuous-delivery` flag:

```
$ ./keptn install --chart-repo=./keptn-0.8.8-dev-PR-4656.tgz  --use-case=continuous-delivery
Error: error retrieving Keptn Helm Chart at ./helm-service-0.8.8-dev-PR-4656.tgz: Get "./helm-service-0.8.8-dev-PR-4656.tgz": unsupported protocol scheme ""
```

Note that with this usecase keptn will try to install the `helm-service` as well as the `jmeter-service` which are alo two separate helm charts.

**Comment by jimil749** [comment]
Gotcha. So I need to add a similar check in the `fetchContinuousDeliveryCharts` function too, so that we fetch `helm-service` and `jmeter-service` charts from the required path. (URL/localfs)

File: cli/cmd/install.go [hunk 2]
```diff
@@ -299,11 +308,20 @@ func (i *InstallCmdHandler) doInstallation(installParams installCmdParams) error
...
 func fetchContinuousDeliveryCharts(helmHelper helm.IHelper, chartRepoURL *string) ([]*chart.Chart, error) {
 	charts := []*chart.Chart{}
+	var serviceChart *chart.Chart
+	var err error
 	for _, service := range continuousDeliveryServices {
 		chartURL := getExecutionPlaneServiceChartRepoURL(chartRepoURL, service)
-		serviceChart, err := helmHelper.DownloadChart(chartURL)
-		if err != nil {
-			return nil, err
+		if goutils.IsValidURL(chartURL) {
```
Comment (warber) [inline cli/cmd/install.go:315]: this code block is 100% the same code as in line `206+`. you can extract that into a separate func.

**Review by warber — CHANGES_REQUESTED** [review]

**Review by warber — APPROVED** [review]
lgtm

**Pull request merged**